Dockerfile

```diff
@@ -1,25 +1,12 @@
-FROM node:14-alpine AS deps
```

Comment:
Thanks @Mogztter, can you please explain a bit how this will make the build faster?

Reply:
Docker will cache every layer, I think it's unnecessary to create a dedicated container to install dependencies. It can be done as part of the build step.

If we want to cache dependencies we could do:

```Dockerfile
FROM node:14-alpine AS build

WORKDIR /opt/node_app
COPY package.json package-lock.json .
RUN npm i --no-optional

# ... rest of the build process
```

If nothing changed, Docker will use previously built layers.


> can you please explain a bit how this will make the build faster

I think it would be faster because we create only one container to build the project and we don't copy files from one container to another using `COPY --from=deps /opt/node_app .`.

Reply:
> If we want to cache dependencies we could do:

Yeah I think that makes more sense, I believe that was the reason for spliting those steps, although I can't recall why we put two Docker stages there. If you could make that change, it would be great, thanks.

Other than that and the package-lock.json change, it looks great, thanks for that. :)